src/kvstore/raftex/RaftPart.cpp

```diff
@@ -479,62 +488,60 @@ void RaftPart::appendLogsInternal(AppendLogsIterator iter, TermID termId) {
                 << iter.logId() << " (Current term is "
                 << currTerm << ")";
     } else {
-        VLOG(2) << idStr_ << "Ready to send a heartbeat";
+        LOG(ERROR) << idStr_ << "Only happend when CAS failed";
+        replicatingLogs_ = false;
+        return;
     }
     AppendLogResult res = AppendLogResult::SUCCEEDED;
     do {
-        {
-            std::lock_guard<std::mutex> g(raftLock_);
-            if (status_ != Status::RUNNING) {
-                // The partition is not running
-                VLOG(2) << idStr_ << "The partition is stopped";
-                res = AppendLogResult::E_STOPPED;
-                break;
-            }
-
-            if (role_ != Role::LEADER) {
-                // Is not a leader any more
-                VLOG(2) << idStr_ << "The leader has changed";
-                res = AppendLogResult::E_NOT_A_LEADER;
-                break;
-            }
-            if (term_ != termId) {
-                VLOG(2) << idStr_ << "Term has been updated, origin "
-                        << termId << ", new " << term_;
-                res = AppendLogResult::E_TERM_OUT_OF_DATE;
-                break;
-            }
-            currTerm = term_;
-            prevLogId = lastLogId_;
-            prevLogTerm = lastLogTerm_;
-            committed = committedLogId_;
-            // Step 1: Write WAL
-            if (!wal_->appendLogs(iter)) {
-                LOG(ERROR) << idStr_ << "Failed to write into WAL";
-                res = AppendLogResult::E_WAL_FAILURE;
-                break;
-            }
-            lastId = wal_->lastLogId();
-            VLOG(2) << idStr_ << "Succeeded writing logs ["
-                    << iter.firstLogId() << ", " << lastId << "] to WAL";
-        }  // end of lock scope
+        std::lock_guard<std::mutex> g(raftLock_);
+        if (status_ != Status::RUNNING) {
```

Comment:
maybe we can package the status/role/term check to a function, because processAppendLogResponses has the same logic.

Reply:
Yes,  but i prepared to remove the retry logic in the future.  Let's leave it here now.